sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts

```diff
@@ -236,54 +238,55 @@ export class AssessmentSequentialProcessingService {
+      // as the overawards are applicable only for the loans and NOT for the grants.
+      .addSelect(
+        "SUM(COALESCE(disbursementValue.effectiveAmount, disbursementValue.valueAmount - COALESCE(disbursementValue.disbursedAmountSubtracted, 0)))",
+        "total",
+      )
       .innerJoin(
-        "currentAssessment.disbursementSchedules",
+        "disbursementValue.disbursementSchedule",
         "disbursementSchedule",
       )
-      .innerJoin("disbursementSchedule.disbursementValues", "disbursementValue")
-      .innerJoin("currentAssessment.offering", "offering")
+      .innerJoin("disbursementSchedule.studentAssessment", "studentAssessment")
+      .innerJoin("studentAssessment.offering", "offering")
+      .innerJoin("studentAssessment.application", "application")
       .where("application.applicationNumber IN (:...applicationNumbers)", {
         applicationNumbers,
       })
       // Overwritten application can have awards associated with and they should not be considered.
       .andWhere("application.applicationStatus != :overwrittenStatus", {
         overwrittenStatus: ApplicationStatus.Overwritten,
       })
-      // Check for assessment completed status to avoid retrieving any cancelation status.
-      .andWhere(
-        "currentAssessment.studentAssessmentStatus = :completedStudentAssessmentStatus",
-        {
-          completedStudentAssessmentStatus: StudentAssessmentStatus.Completed,
-        },
-      )
       // Only consider disbursements in Pending, ReadyToSend, or Sent.
       .andWhere(
-        "disbursementSchedule.disbursementScheduleStatus != :cancelledDisbursementScheduleStatus",
+        "disbursementSchedule.disbursementScheduleStatus IN (:...payableDisbursementScheduleStatus)",
         {
-          cancelledDisbursementScheduleStatus:
-            DisbursementScheduleStatus.Cancelled,
+          payableDisbursementScheduleStatus: [
+            DisbursementScheduleStatus.Pending,
+            DisbursementScheduleStatus.ReadyToSend,
+            DisbursementScheduleStatus.Sent,
+          ],
         },
       )
-      // Sequenced applications need at least one valid COE, which means that one can be cancelled
-      // and the other still valid. This ensures that only the valid one will be considered.
-      .andWhere("disbursementSchedule.coeStatus != :declinedCOEStatus", {
-        declinedCOEStatus: COEStatus.declined,
-      })
       // Ensures that only grants will be returned since loans and not needed.
       .andWhere("disbursementValue.valueType IN (:...grantsValueTypes)", {
         grantsValueTypes: [
           DisbursementValueType.CanadaGrant,
           DisbursementValueType.BCGrant,
         ],
       })
-      .groupBy("offering.offeringIntensity")
-      .addGroupBy("disbursementValue.valueCode")
+      .groupBy("disbursementValue.valueCode")
+      .addGroupBy("offering.offeringIntensity")
```

Comment:
Was there any benefit of switching the `groupBy` order?

Reply:
None. When I had re-written the query, order just happened to change. I can change it back to avoid any confusion or leave it as it is.